packages/flutter/lib/src/gestures/scale.dart

```diff
@@ -247,13 +247,22 @@ class _LineBetweenPointers{
 class ScaleGestureRecognizer extends OneSequenceGestureRecognizer {
   /// Create a gesture recognizer for interactions intended for scaling content.
   ///
-  /// {@macro flutter.gestures.GestureRecognizer.kind}
+  /// {@macro flutter.gestures.GestureRecognizer.supportedDevices}
   ScaleGestureRecognizer({
     Object? debugOwner,
+    @Deprecated(
+      'Migrate to supportedDevices. '
+      'This feature was deprecated after v2.3.0-1.0.pre.',
+    )
     PointerDeviceKind? kind,
+    Set<PointerDeviceKind>? supportedDevices,
     this.dragStartBehavior = DragStartBehavior.down,
   }) : assert(dragStartBehavior != null),
-       super(debugOwner: debugOwner, kind: kind);
+       assert(kind == null || supportedDevices == null),
```

Comment:
Can you rely on the assert in the superclass by passing the arguments straight through (unmodified) to `super()`?

(here and elsewhere)

Reply:
Error messages (and accompanying stack traces) may be slightly clearer if the assert happens as early as possible.

Reply:
I was going back and forth on exactly that. Earlier error/stack trace or simpler code... I'm happy to do either. Curious what folks think about the either/or.
Since the parameters are the same for the super and sub classes (kind/supportedDevices), maybe an error on the super class would not be that obscure @goderbauer WDYT?